### PR TITLE
update id.k8s.io claim if empty

### PR DESCRIPTION
**Pull request by zhiweiyin318**
Signed-off-by: Zhiwei Yin <zyin@redhat.com>

issue :  https://issues.redhat.com/browse/ACM-2998

### CODE REVIEW

File: pkg/klusterlet/clusterclaim/clusterclaim_controller.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func (r *ClusterClaimReconciler) createOrUpdate(ctx context.Context, newClaim *c
 		return fmt.Errorf("unable to get ClusterClaim %q: %w", newClaim.Name, err)
 	case !reflect.DeepEqual(oldClaim.Spec, newClaim.Spec):
 		// if newClaim.Name is in clusterClaimCreateOnlyList, then do nothing
-		if utils.ContainsString(clusterClaimCreateOnlyList, newClaim.Name) {
+		if utils.ContainsString(clusterClaimCreateOnlyList, newClaim.Name) && oldClaim.Spec.Value != "" {
```
Comment (qiujian16) [inline pkg/klusterlet/clusterclaim/clusterclaim_controller.go:100]: do you need to also ensure newClaim is not empty?

Comment (zhiweiyin318) [reply pkg/klusterlet/clusterclaim/clusterclaim_controller.go:100]: check in https://github.com/stolostron/multicloud-operators-foundation/pull/552/files#diff-c78e2a145aa414b25e54617166eced7480b482f0683ae9736ece7497acdbe2d5R175 
if clusterID is empty, will not newClaim

Comment (zhiweiyin318) [reply pkg/klusterlet/clusterclaim/clusterclaim_controller.go:103]: added a check in syncClaims before call createOrUpdate

Comment (zhiweiyin318) [reply pkg/klusterlet/clusterclaim/clusterclaim_controller.go:100]: @elgnay  do we have empty value claim scenario? I found there are empty value claim case in the unit tests.

Comment (zhiweiyin318) [reply pkg/klusterlet/clusterclaim/clusterclaim_controller.go:100]: In this case, for the clam in clusterClaimCreateOnlyList , do not need to update when the old claim has valid value.

**Comment by zhiweiyin318** [comment]
cherry-pick /backplane-2.2

**Pull request merged**